meilisearch-lib/src/index/search.rs

```diff
@@ -97,8 +102,13 @@ impl Index {
             search.query(query);
         }
 
-        search.limit(query.limit);
-        search.offset(query.offset.unwrap_or_default());
+        // Make sure that a user can't get more documents than the hard limit,
+        // we align that on the offset too.
+        let offset = min(query.offset.unwrap_or(0), HARD_RESULT_LIMIT);
```

Comment:
```suggestion
        let offset = query.offset.unwrap_or(0).min(HARD_RESULT_LIMIT));
```

Reply:
I find it way easier to use the `cmp::min` function like I do than using the `usize::min` method.